src/lib.rs

```diff
@@ -53,189 +34,67 @@ extern crate rayon;
-    ///
-    /// ```rust
-    /// # use tsunami::{TsunamiBuilder, MachineSetup};
-    /// let mut b = TsunamiBuilder::default();
-    /// b.add_set(
-    ///     "server",
-    ///     1,
-    ///     MachineSetup::new("m5.large", "ami-e18aa89b", |ssh| {
-    ///         ssh.cmd("yum install nginx").map(|out| {
-    ///             println!("{}", out);
-    ///         })
-    ///     }),
-    /// );
-    /// b.add_set(
-    ///     "client",
-    ///     10,
-    ///     MachineSetup::new("m5.large", "ami-e18aa89b", |ssh| {
-    ///         ssh.cmd("yum install wget").map(|out| {
-    ///             println!("{}", out);
-    ///         })
-    ///     }),
-    /// );
-    /// ```
-    pub fn add_set(&mut self, name: &str, number: u32, setup: MachineSetup) {
-        // TODO: what if name is already in use?
-        self.descriptors.insert(name.to_string(), (setup, number));
-    }
-
-    /// Set up the machines in a specific EC2
-    /// [`Region`](http://rusoto.github.io/rusoto/rusoto_core/region/enum.Region.html).
-    ///
-    /// The default region is us-east-1. [Available regions are listed
-    /// here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-available-regions)
-    pub fn set_region(&mut self, region: Region) {
-        self.region = region;
-    }
-
-    /// Set up the machines in a specific EC2 availability zone.
-    ///
-    /// This controls the `availability_zone` field of the
-    /// [`SpotPlacement`](https://rusoto.github.io/rusoto/rusoto_ec2/struct.SpotPlacement.html)
-    /// struct (N.B.: even though the documentation claims that the parameter only affects spot
-    /// fleets, it does appear to affect *all* spot instances).
-    pub fn set_availability_zone(&mut self, zone: &str) {
-        self.availability_zone = Some(zone.to_string());
-    }
-
-    /// By default, all spawned instances are launched in a single [Placement
-    /// Group](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html) using the
-    /// [cluster](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html#placement-groups-cluster)
-    /// policy. This ensures that all the instances are located in the same availability region,
-    /// and in close proximity to one another.
-    ///
-    /// This places [some
-    /// restrictions](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html#concepts-placement-groups)
-    /// on the launched instances, and causes it to be more likely for placements to fail. Call
-    /// this method to disable clustering. This will leave instance placing entirely up to ec2,
-    /// which may choose to place your instances in disparate availability zones.
-    pub fn no_clustering(&mut self) {
-        self.cluster = false;
+impl<L: Launcher> TsunamiBuilder<L> {
+    /// Add a machine descriptor to the Tsunami.
+    /// Machine descriptors are specific to the cloud provider they will be used for.
```

Comment:
You probably want to leave an extra newline after the first doc line. It causes rustdoc to realize that the first line is a summary (I believe).